File: adapter-pattern/java/src/MediaPlayer.java
package src;

// MediaPlayer支持视频/音频的格式播放接口，作为基础接口。
public interface MediaPlayer {
  String fileType = null;
  String filePath = null;
  public void play();
  public void setFileType(String fileType);
  public void setFilePath(String filePath);
}

File: adapter-pattern/python/src/MediaPlayer.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""

class MediaPlayer():

    def __init__(self, file_type, file_path):
        self.file_type = file_type
        self.file_path = file_path

    def play(self, ):
        print('MediaPlayer fileType:' + self.file_type + ' | filePath:' +
              self.file_path)
        print('MediaPlayer::play()')

    def set_file_type(self, file_type):
        self.file_type = file_type

    def set_file_path(self, file_path):
        self.file_path = file_path
